### PR TITLE
DEVEXP-294: Create variations for methods of verification

### PR DESCRIPTION
**Pull request by asein-sinch**
- create 4 variations of start verifications (sms, callout, flashCall, seamless)
 - create 3 variations of report by Id (sms, callout, flashCall)
 - create 3 variations of report by identity (sms, callout, flashCall)
 - register the callback service onto the verification controller
 - clean models

### CODE REVIEW

File: packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts [hunk 1]
```diff
@@ -1,18 +1,59 @@
...
+
+  /**
+   * Validate authorization header for callback request
+   * @param {IncomingHttpHeaders} headers - Incoming request's headers
+   * @param {string} path - Incoming request's path
+   * @param {any} body - Incoming request's body
+   * @param {string} method - Incoming request's HTTP method
+   * @return {boolean} - true if the authorization header is valid
+   */
+  public validateAuthorizationHeader(
```
Comment (JPPortier) [inline packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts:23]: `validateAuthenticationHeader` as common public interface ?

File: packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts [hunk 2]
```diff
@@ -1,18 +1,59 @@
...
-  throw new Error('Unknown Verification event');
-};
+
+  /**
+   * Reviver for a Verification Event.
+   * This method ensures the object can be treated as a Verification Event and should be called before any action is taken to manipulate the object.
+   * @param {any} eventBody - The event body containing the verification event notification.
+   * @return {VerificationCallback} - The parsed verification event object.
+   */
+  public parseVerificationEventNotification(eventBody: any): VerificationCallback {
```
Comment (JPPortier) [inline packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts:44]: `parseEvent()` as common public interface ?

File: packages/verification/src/models/v1/helper.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    identity: string,
+    code: string,
+    cli?: string,
+  ): ReportSmsVerificationByIdentityRequestData => {
+    return {
+      endpoint: identity,
+      reportSmsVerificationByIdentityRequestBody: {
+        sms: {
+          code,
+          cli,
```
[the remark this answers is not in the document]
Comment (asein-sinch) [reply packages/verification/src/models/v1/helper.ts:127]: I was sure you would note that and I have written a test especially for you:
 - [without CLI](https://github.com/sinch/sinch-sdk-node/pull/21/files#diff-978e0c8b1a22ba431c1418f7a2025ba61faa2da08cdb6944d46f0d0559ee00f9R124)
 - [with CLI](https://github.com/sinch/sinch-sdk-node/pull/21/files#diff-978e0c8b1a22ba431c1418f7a2025ba61faa2da08cdb6944d46f0d0559ee00f9R136)

File: packages/verification/src/models/v1/start-seamless-verification-response/start-seamless-verification-response.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export interface StartSeamlessVerificationResponse {
+
+  /** Verification identifier used to query for status. */
+  id: string;
+  /** The value of the method used for the Verification. For Data Verifications, this will always be `seamless`. */
+  method?: 'seamless';
```
[the remark this answers is not in the document]
Comment (asein-sinch) [reply packages/verification/src/models/v1/start-seamless-verification-response/start-seamless-verification-response.ts:6]: This is what the specification says, I don't know if the server will always send it or not 🤷‍♂️

File: packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts (hunk 1 above)
Comment (asein-sinch) [reply packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts:23]: Will be refactored in the PR#22

File: packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts (hunk 2 above)
Comment (asein-sinch) [reply packages/verification/src/rest/v1/callbacks/callbacks-webhook.ts:44]: Will be refactored in the PR#22

**Review by JPPortier — APPROVED** [review]

**Review by 650elx — APPROVED** [review]

**Pull request merged**